main/tests/server/src/com/google/refine/grel/controls/FilterTests.java

```diff
@@ -27,15 +27,53 @@
 
 package com.google.refine.grel.controls;
 
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.util.Properties;
+
 import org.testng.annotations.Test;
 
+import com.google.refine.expr.EvalError;
+import com.google.refine.expr.Evaluable;
+import com.google.refine.expr.MetaParser;
+import com.google.refine.expr.ParsingException;
+import com.google.refine.grel.GrelTestBase;
 import com.google.refine.util.TestUtils;
 
-public class FilterTests {
+public class FilterTests extends GrelTestBase {
 
     @Test
     public void serializeFilter() {
         String json = "{\"description\":\"Evaluates expression a to an array. Then for each array element, binds its value to variable name v, evaluates expression test which should return a boolean. If the boolean is true, pushes v onto the result array.\",\"params\":\"expression a, variable v, expression test\",\"returns\":\"array\"}";
         TestUtils.isSerializedTo(new Filter(), json);
     }
+
+    private void assertParseError(String expression) throws ParsingException {
```

Comment:
@wetneb As the 1st case is a positive case; i updated the name of the function and its working

Reply:
Well, now this first case does not assert that the returned value is an `EvalError` anymoe, which I think would be worth doing. So I would add back this assertion and change the name of this method to `assertEvaluatesToError`, again.

You seem to be unclear on the distinction between a parse error and an evaluation error. Let me try to explain this distinction with an analogy: instead of talking about GREL expressions, let's talk about sentences in English.
Consider the following sentences:
1. "The moon satellite a is of Elvis Presley." - > this sentence is syntactically incorrect: it has a parse error, because the words are in an incorrect order
2. "The moon is a satellite of Elvis Presley." -> this sentence is syntactically correct, but it does not make sense: humans don't have satellites.
3. "The moon is a satellite of the Earth." -> this sentence is syntactically correct, and its meaning is true.

As you can see in those examples, sentences in English can have (at least) two sorts of problems:
1. being syntactically incorrect: that means that their structure does not respect the grammar of the language
2. being semantically wrong: that means that although their structure respects the rules of the language, they don't make sense

Similarly, in GREL, expressions can have two sorts of problems:
* a syntax error, meaning that they do not conform to the grammar of GREL. In this case, `ParsingException` is thrown
* an evaluation error: meaning that although they could be parsed successfully (without throwing a `ParsingException`), evaluating them returns an `EvalError`

When writing tests for this, it is important that this distinction is clear so that you can correctly assert whether a `ParsingException` is thrown at the parsing stage or an `EvalError` is returned at the evaluation stage.

Reply:
@wetneb Thanks for the explanation; i missed the Evaluation error case. I have updated the function. Please review